### PR TITLE
Wrapper / implementation for endroid/qr-code

### PR DESCRIPTION
**Pull request by julianzimmermann**
- added epc qr code service
- added plugin skeleton

### CODE REVIEW

File: bundles/payment-epc-qr-code/composer.json [hunk 1]
```diff
@@ -9,9 +9,11 @@
     }
   ],
   "require": {
-    "php": ">=7.3"
+    "php": ">=7.3",
+    "endroid/qr-code": "^4.1.0"
   },
   "require-dev": {
+    "endroid/qr-code": "^4.1",
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/composer.json:16]: It's duplicate...

File: bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/Dependency/Wrapper/QrCodeWrapper.php [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            case 3:
+                $errorLevel = new ErrorCorrectionLevelQuartile();
+
+                break;
+            default:
+                throw new Exception(sprintf(
+                    'Error level %s not known. Please chose from 0 to 3 (low, medium, high, quartile)',
+                    $this->config->getErrorCorrectionLevel()
+                ));
+        }
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/Dependency/Wrapper/QrCodeWrapper.php:107]: Maybe a hash map (assoc array) will be better.

File: bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/Dependency/Wrapper/QrCodeWrapper.php [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            case 3:
+                $mode = new RoundBlockSizeModeShrink();
+
+                break;
+            default:
+                throw new Exception(sprintf(
+                    'Mode %s not known. Please chose from 0 to 3 (none, margin, large, shrink)',
+                    $this->config->getErrorCorrectionLevel()
+                ));
+        }
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/Dependency/Wrapper/QrCodeWrapper.php:143]: Same...

File: bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/Dependency/Wrapper/QrCodeWrapper.php [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+        $usage = $paymentEpcQrCodeRequestTransfer->getUsage() ?? '';
+        $reference = $paymentEpcQrCodeRequestTransfer->getReference() ?? '';
+        $purpose = $paymentEpcQrCodeRequestTransfer->getPurpose() ?? '';
+
+        if (empty($usage) && empty($reference)) {
+            throw new Exception(sprintf('One of "Reference(%s)" or "Purpose(%s)" must be null!', $paymentEpcQrCodeRequestTransfer->getReference(), $paymentEpcQrCodeRequestTransfer->getUsage()));
+        }
+
+        return $paymentEpcQrCodeRequestTransfer->getServiceTag() . PHP_EOL .
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/Dependency/Wrapper/QrCodeWrapper.php:198]: Maybe sprintf will be better...

File: bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/PaymentEpcQrCodeDependencyProvider.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    /**
+     * @param \Spryker\Service\Kernel\Container $container
+     *
+     * @return \Spryker\Service\Kernel\Container
+     */
+    public function addQrCodeWrapper(Container $container): Container
+    {
+        $container[static::WRAPPER_QR_CODE] = function () {
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/src/FondOfOryx/Service/PaymentEpcQrCode/PaymentEpcQrCodeDependencyProvider.php:35]: Use static closure...

File: bundles/payment-epc-qr-code/composer.json [hunk 6]
```diff
@@ -9,9 +9,11 @@
     }
   ],
   "require": {
-    "php": ">=7.3"
+    "php": ">=7.3",
+    "endroid/qr-code": "^4.1.0"
   },
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/composer.json:14]: Did you require all needed packages? You use a plugin contract from oms-extension.

File: composer.json [hunk 7]
```diff
@@ -15,6 +15,7 @@
   "require": {
     "php": ">=7.3",
     "ext-json": "*",
+    "endroid/qr-code": "^4.1",
```
Comment (daniel-rose) [inline composer.json:18]: Use in require section only version constraints like "^MAJOR.MINOR.PATCH"

**Review by daniel-rose — APPROVED** [review]

**Review by gengjozsef — APPROVED** [review]

**Pull request merged**